.flake8

```diff
@@ -1,5 +1,5 @@
 [flake8]
-ignore = D100,D103,D104
+extend-ignore = D100,D103,D104
```

Comment:
Can you explain this change? There's no context at all. Did it change upstream or is this an alternative way of expressing the ignore list or ...?

Reply:
Since we applied the same fix in various other repos in the past I don't add more context. From the `flake8` usage:

>  --ignore=errors       Comma-separated list of errors and warnings to ignore
>                        (or skip). For example, ``--ignore=E4,E51,W234``.
>                        (Default: E121,E123,E126,E226,E24,E704,W503,W504)
>  --extend-ignore=errors
>                        Comma-separated list of errors and warnings to add to
>                        the list of ignored ones. For example, ``--extend-
>                        ignore=E4,E51,W234``.

So by using `ignore` we overwrite the code ignored by default. There are some error codes in `flake8` which are contradicting each other which is why some of them are ignored by default. By using `extend-ignore` we only contribute to the list without resetting it.

Reply:
> Since we applied the same fix in various other repos in the past

It would be nice to link to one, otherwise the reviewer has no context unless they have reviewed one before or unless they have read every change that has come in the past.

> So by using ignore we overwrite the code ignored by default. There are some error codes in flake8 which are contradicting each other which is why some of them are ignored by default. By using extend-ignore we only contribute to the list without resetting it.

Ok, makes sense. Thanks for the explanation, but it should probably be linked to if we need to make a similar change in the future.

Reply:
> it should probably be linked to if we need to make a similar change in the future.

* https://github.com/ros-infrastructure/ros_buildfarm/pull/611#discussion_r267617889
* ament/ament_lint#122
* dirk-thomas/vcstool#79
* colcon/colcon-core#127